internal/driver/subscriptionrequest_test.go

```diff
@@ -0,0 +1,45 @@
+package driver
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestParseISO8601 verifies accurate parsing of time.
+func TestParseISO8601(t *testing.T) {
+	tests := []struct {
+		input         string
+		expected      time.Duration
+		errorExpected bool
+	}{
+		{
+			input:    "PT180S",
+			expected: 180000000000,
+		},
+		{
+			input:    "P1Y2M3W4DT5H6M7S",
+			expected: 18367000000000,
```

Comment:
i would add more test cases that skips values in between since that is part of the support, like 1 year 5 hours. 5 days, 2 minutes. etc.